objectivec/include/ort_session.h

```diff
@@ -46,14 +54,151 @@ NS_ASSUME_NONNULL_BEGIN
  *
  * @param inputs Dictionary of input names to input ORT values.
  * @param outputNames Set of output names.
+ * @param runOptions Run configuration options.
  * @param[out] error Optional error information set if an error occurs.
  * @return A dictionary of output names to output ORT values with the outputs
  *         requested in `outputNames`, or nil if an error occurs.
  */
 - (nullable NSDictionary<NSString*, ORTValue*>*)runWithInputs:(NSDictionary<NSString*, ORTValue*>*)inputs
                                                   outputNames:(NSSet<NSString*>*)outputNames
+                                                   runOptions:(ORTRunOptions*)runOptions
                                                         error:(NSError**)error;
 
 @end
 
+/**
+ * Options for configuring a session.
+ */
+@interface ORTSessionOptions : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates session configuration options.
+ *
+ * @param[out] error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithError:(NSError**)error NS_SWIFT_NAME(init());
+
+/**
+ * Sets the number of threads used to parallelize the execution within nodes.
+ * A value of 0 means ORT will pick a default value.
+ *
+ * @param intraOpNumThreads The number of threads.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setIntraOpNumThreads:(int)intraOpNumThreads
+                       error:(NSError**)error;
+
+/**
+ * Sets the graph optimization level.
+ *
+ * @param graphOptimizationLevel The graph optimization level.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setGraphOptimizationLevel:(ORTGraphOptimizationLevel)graphOptimizationLevel
+                            error:(NSError**)error;
+
+/**
+ * Sets the path to which the optimized model file will be saved.
+ *
+ * @param optimizedModelFilePath The optimized model file path.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setOptimizedModelFilePath:(NSString*)optimizedModelFilePath
+                            error:(NSError**)error;
+
+/**
+ * Sets the session log ID.
+ *
+ * @param logID The log ID.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setLogID:(NSString*)logID
+           error:(NSError**)error;
+
+/**
+ * Sets the session log severity level.
+ *
+ * @param loggingLevel The log severity level.
+ * @param[out] error Optional error information set if an error occurs.
+ * @return Whether the option was set successfully.
+ */
+- (BOOL)setLogSeverityLevel:(ORTLoggingLevel)loggingLevel
+                      error:(NSError**)error;
+
```

Comment:
Side note: Do we need to add a logging sink that makes sense on iOS or is the default std::clog output fine? On Android I believe we hook into a system specific logger.

Reply:
We will need to hook up to Apple's logging framework, to have best backward compatiblities, need to hook up a NSLog sink

Reply:
thanks for bringing it up. I can look into it, probably for a future PR

Reply:
yeah - definitely a separate change